### DISCUSSION TITLE
delay messages stucks in exchange while node restarted after crashdown

### DISCUSSION THREAD

**Discussion by liolay** [rabbitmq/rabbitmq-delayed-message-exchange#289] [General]
### Describe the bug

delay messages stucks in exchange while node restarted after crashdown, all delayed message can't routing to bind queues,  just stucks in delay exchange, but at the same time, normal messages is fine, to make this plugin works ,we must stop and remove data dir and restart rabbitmq

### Reproduction steps

there is a probability of occurrence when a node crashes

### Expected behavior

this plugin works fine

**Comment by michaelklishin** [comment]
It is impossible to suggest anything specific. You haven't shared any versions details, logs or steps to reproduce.

I don't know what a "crashdown" is but this plugin has documented limitations, for example, it stores messages in a node-local Mnesia table.

**Reply by liolay** [reply]
"crashdown" means unexpected power outage on server. we run rabbitmq:3.10.25-management on docker container, container can start success and nonthing error logs output, while, delayed message already saved in Mnesia table and newly created delayed message stucks in Mnesia even they expired. i have sent "docker volume data dir" to your mail "[michael@clojurewerkz.org](mailto:michael@clojurewerkz.org)". mount this to rabbitmq can reproduce problem

**Reply by liolay** [reply]
aha~ mail was backed..., is there any other way to do this (maybe contains sensitive data, can't just follow in github)?

**Reply by michaelklishin** [reply]
What you have sent me is the node's data directory, not logs. However, I see that the plugin version is 3.10. RabbitMQ 3.10.x has been out of support for years.

**Comment by michaelklishin** [comment]
@liolay this is discussion material => moved to a discussion.

**Comment by gomoripeti** [comment]
You might face this issue https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/pull/214 which was addressed in 3.12 version of the plugin

**Reply by liolay** [reply]
Wow! this is it! Unfortunately, we have already used DLQ as an alternative solution. 
Anyway, thanks!
